src/Serializer/AbstractItemNormalizer.php

```diff
@@ -313,9 +320,9 @@ protected function denormalizeRelation(string $attributeName, PropertyMetadata $
 
         if (!\is_array($value)) {
             // repeat the code so that IRIs keep working with the json format
-            if (true === $this->allowPlainIdentifiers && $this->itemDataProvider) {
+            if (true === $this->allowPlainIdentifiers && $this->iriConverter instanceof IriPlainIdentifierAwareConverterInterface) {
                 try {
-                    return $this->itemDataProvider->getItem($className, $value, null, $context + ['fetch_data' => true]);
+                    return $this->iriConverter->getItemFromIri($this->iriConverter->getIriFromPlainIdentifier($value, $className), $context + ['fetch_data' => true]);
                 } catch (ItemNotFoundException $e) {
                     throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
                 } catch (InvalidArgumentException $e) {
```

Comment:
I think that we're missing the:

```
                   if (null === $item) {
                        throw new ItemNotFoundException(sprintf('Item with id "%s" not found for class "%s".', $value, $className));
                    }
```

Reply:
We don't need that, because getItemFromIri already throws